docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_encrypted.mdx

```diff
@@ -23,6 +23,219 @@ HIPAA, GDPR, NIST, SOC2, HITRUST, PCIDSS, NISTCSF
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Set up a Boto3 session with the necessary AWS credentials and region.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   redshift = session.client('redshift')
+   ```
+
+3. **Check Existing Clusters for Encryption**:
+   Write a function to check if existing Redshift clusters are encrypted. If not, you can log a warning or take necessary actions.
+   ```python
+   def check_redshift_encryption():
+       clusters = redshift.describe_clusters()
+       for cluster in clusters['Clusters']:
+           if not cluster['Encrypted']:
+               print(f"Cluster {cluster['ClusterIdentifier']} is not encrypted.")
+           else:
+               print(f"Cluster {cluster['ClusterIdentifier']} is encrypted.")
+
+   check_redshift_encryption()
+   ```
+
+4. **Create Encrypted Redshift Cluster**:
+   When creating a new Redshift cluster, ensure that the `Encrypted` parameter is set to `True`.
+   ```python
+   def create_encrypted_redshift_cluster(cluster_identifier, node_type, master_username, master_user_password, db_name):
+       response = redshift.create_cluster(
+           ClusterIdentifier=cluster_identifier,
+           NodeType=node_type,
+           MasterUsername=master_username,
+           MasterUserPassword=master_user_password,
+           DBName=db_name,
+           Encrypted=True
+       )
+       print(f"Creating encrypted cluster {cluster_identifier}.")
+       return response
+
+   # Example usage
+   create_encrypted_redshift_cluster(
+       cluster_identifier='my-encrypted-cluster',
+       node_type='dc2.large',
+       master_username='admin',
+       master_user_password='YourPassword123',
+       db_name='mydatabase'
+   )
+   ```
+
+By following these steps, you can ensure that your Redshift clusters are encrypted, thereby preventing misconfigurations related to unencrypted clusters.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Encryption" column. If the value is "Disabled" or "Not enabled", then the Redshift cluster is not encrypted.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access Redshift resources.
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>